proposals/features-library.md

```diff
@@ -0,0 +1,76 @@
+# Sharing code between Dev Container Features
+
+ref: https://github.com/devcontainers/spec/issues/129
+
+# Motivation
+
+The current model for authoring Features encourages significant copy and paste between individual Features.  Most of the Features published under the `devcontainers/features` namespace contain approximately 100 lines of [identifical helper functions](https://github.com/devcontainers/features/blob/main/src/azure-cli/install.sh#L29-L164) before the first line of the script is executed.  The process of maintaining these helper functions is tedious, error-prone, and leads to less readable Features.
+
+Additionally, [members of the Feature authoring community have expressed interest in sharing code between Features](https://github.com/orgs/devcontainers/discussions/10).  Exploring the source code of self-published Features, it is evident that many authors have copy/pasted code from other Features.  This is a sign that there is a need for a more efficient way to share code between Features.
+
+# Goal
+
+Provide a generic and efficient way to share code between Features.
+
+# Proposal
+
+Add an `include` property to the `devcontainer-feature.json`.  This will be honored during the Feature [packaging stage](https://containers.dev/implementors/features-distribution/#packaging). The contents of the provided relative path is copied into the packaged Feature archive.
+
+Property | Type | Description
+--- | --- | ---
+`include` | `string[]` | An array of relative paths to package with the Feature. Paths are relative to root directory when packaging (often the git root directory).  If the element is a folder, it is copied recursively.
```

Comment:
What would happen for paths that resolve outside of the feature root directory, such as `../shared/utils/`?

Reply:
Similar to how we don't resolve Local Features outside of the `.devcontainer` folder, I propose failing the packaging step.

Reply:
Added more detail